### PR TITLE
SDK-86 Install Open Web App

### PR DESCRIPTION
**Pull request by PawelGutkowski**
This PR introduces changes:

1. add `Bintray` class to interact with Bintray's Rest API and `OpenmrsBintray` to easily obtain packages from openmrs repositories on Bintray
2. add `Deploy OpenMRS OWA` option to `deploy` task  in interactive and batch mode (eg parameter `-Dowa=conceptdictionary:1.0.0-beta.6`)
3. store OWAs installed by user in server properties
4. create appropriate integration tests

### CODE REVIEW

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/Deploy.java [hunk 1]
```diff
@@ -151,13 +160,51 @@ private void runInteractiveMode(Server server, ServerUpgrader upgrader) throws M
...
+            for(BintrayId id : bintray.getAvailableOWA()){
+                owas.add(id.getName());
+            }
+            name = wizard.promptForMissingValueWithOptions("Which OWA would you like to deploy?%s", name, "", owas, true);
+        }
+        if(version == null){
+            List<String> versions = bintray.getOwaMetadata(name).getVersions();
+            version = wizard.promptForMissingValueWithOptions("Which version would you like to deploy?%s", version, "", versions, true);
+        }
+        wizard.showMessage("Please remember that to run OWA you need to have openmrs-owa-module installed on the server");
```
Comment (rkorytkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/Deploy.java]: Could we actually check if it is installed here and if not install the latest version?

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/Deploy.java [hunk 2]
```diff
@@ -151,13 +160,51 @@ private void runInteractiveMode(Server server, ServerUpgrader upgrader) throws M
...
+        File owaDir = new File(server.getServerDirectory(), "owa");
+        if(!owaDir.exists()){
+            //couldn't find place where owa dir is saved yet
+            boolean useDefaultDir = wizard.promptYesNo(String.format(
+                    "\nThere is no default directory '%s' on server %s, would you like to create it? (if not, you will be asked for path to custom directory)",
```
Comment (rkorytkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/Deploy.java]: Could you please explain this to me? Why would one want create a custom directory?

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/bintray/Bintray.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import org.apache.commons.httpclient.HttpClient;
+import org.apache.commons.httpclient.methods.GetMethod;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+
+public class Bintray {
```
Comment (rkorytkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/bintray/Bintray.java]: I like that you created a wrapper!

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/bintray/Bintray.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    public static List<BintrayId> getAvailablePackages(String owner, String repo){
```
Comment (rkorytkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/bintray/Bintray.java]: Any reason those are static? Statics are harder to test/mock and can't be overriden when extending.

**Comment by rkorytkowski** [comment]
Please address my comments in a separate PR.

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/Deploy.java (hunk 2 above)
Comment (PawelGutkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/Deploy.java]: There is option in OWA module to use custom app directory, so I thought we should support that possibility.

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/bintray/Bintray.java (hunk 4 above)
Comment (PawelGutkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/bintray/Bintray.java]: I thought that would make sense, as there is no state preserved here, but factors You listed seem more important. I will fix this and adress Your other comments in follow-up PR, thanks!

**Pull request merged**